src/routing/Router.svelte

```diff
@@ -29,42 +27,22 @@
   }
 
   // We need to reference $currentQuery in this block if we want the block to
-  // re-run whenever $currentQuery changes. If we were to call
-  // updateQueryString() with no arguments and use $currentQuery directly in the
-  // implementation of updateQueryString(), the query string would not update
-  // when the user uses the menu buttons (Channel, OS, etc.), for example.
+  // run whenever $currentQuery changes. If we were to call updateQueryString()
+  // with no arguments and use $currentQuery in the implementation of
+  // updateQueryString(), this block would never run and the query string would
+  // therefore never update in response to the user activity.
   //
-  // We also need to check that the user is authenticated before doing this,
-  // otherwise we risk clobbering the "code" and "state" query parameters that
-  // the Auth0 client needs to read to authenticate the user. (Strictly
-  // speaking, we don't need to check $store.auth.isAuthenticated here because
-  // it should always be true when handledAuthRedirect is true, but it doesn't hurt
-  // to double check.)
-  $: if (handledAuthRedirect && $store.auth.isAuthenticated && visible) {
+  // We also need to avoid running this block whenever the code= and state= keys
+  // are in the query, otherwise we would clobber them. Auth0 needs to read them
+  // to authenticate the user.
+  $: if (
+    visible
+    && !window.location.search.includes('code=')
```

Comment:
this feels brittle but i see it was always present

Reply:
How do you mean? I'm happy to improve it. 😃